tests/e2e/crashlytics/crashlytics.e2e.js

```diff
@@ -27,9 +27,14 @@ describe('crashlytics()', () => {
       }
     });
 
-    xit('should error on a non a string value', async () => {
-      // TODO lib needs validations adding
-      await firebase.crashlytics().log(123456);
+    it('should error on a non string value', async () => {
+      if (device.getPlatform() !== 'ios') {
+        try {
+          await firebase.crashlytics().log(123456);
+        } catch (e) {
```

Comment:
out of curiousity - shouldn't all of these contain some sort of 'fail' assertion right after the await in the try block, before the catch, to assert that it really did error since it should never get past the await? I've done lots of java but not much javascript testing so I might be off base, but in java testing I'd put a fail after the line that should throw, to make sure I wasn't just silently passing through everything...

Reply:
If my memory serves correct, in JS if you don't catch the throw it will automatically fail out as an uncaught exception.

Reply:
Nothing else jumped out at me - I'll read again and check locally for android e2e, I asked Salakar about the CLA thing (he has the admin oomph to fix that)

Reply:
> If my memory serves correct, in JS if you don't catch the throw it will automatically fail out as an uncaught exception.

Hmm - but what happens in these structures as they are (with proper try/catch) if the method call doesn't throw? wouldn't it silently pass even though we expect an exception? (at least for the ones where we are purposely sending in junk, expecting it to throw)

Reply:
I see where you're going with that, that's why it took me so long to rewrite the tests, I had to make sure on the forced errors that I was actually passing the test on the error call and not on the junk args.

I had a ton of console.log calls to verify everything going in and coming out when I was reworking the test suite.

Reply:
Okay - that makes sense but does seem like a Java / Javascript difference I'll have to get over personally ;-). Ok, I'll check this via android / e2e here and re-read etc then ideally merge it in. 💪